src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -18,28 +19,37 @@
     public static final String COMMAND_WORD = "delete";
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
-            + ": Deletes the study spot identified by the index number used in the displayed study spot list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + ": Deletes the study spot identified by the study spot name used in the displayed study spot list.\n"
+            + "Parameters: Study Spot Name \n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_DELETE_SPOT + "starbucks";
 
     public static final String MESSAGE_DELETE_STUDYSPOT_SUCCESS = "Deleted study spot: %1$s";
 
-    private final Index targetIndex;
+    private Name name;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(Name name) {
+        this.name = name;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<StudySpot> lastShownList = model.getFilteredStudySpotList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_STUDYSPOT_DISPLAYED_INDEX);
+        boolean isPresent = false;
+        StudySpot studySpotToDelete = null;
+        for (StudySpot current: lastShownList) {
+            if (current.isSameName(name)) {
+                studySpotToDelete = current;
+                isPresent = true;
+                break;
+            }
+        }
+
+        if (!isPresent) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EDIT_NAME);
```

Comment:
Could you refactor the name of MESSAGE_INVALID_EDIT_NAME to MESSAGE_INVALID_NAME instead?
If you think it makes more sense